common/src/main/java/org/conscrypt/HpkeSuite.java

```diff
@@ -0,0 +1,330 @@
+  private final KDF mKdf;
+  private final AEAD mAead;
+
+  private HpkeSuite(KEM Kem, KDF Kdf, AEAD Aead) {
+    mKem = Kem;
+    mKdf = Kdf;
+    mAead = Aead;
+  }
+
+  /**
+   * KEM configured while creating an instance of {@link HpkeSuite}
+   *
+   * @return kem
+   */
+  KEM getKem() {
+    return mKem;
+  }
+
+  /**
+   * KDF configured while creating an instance of {@link HpkeSuite}
+   *
+   * @return kdf
+   */
+  KDF getKdf() {
+    return mKdf;
+  }
+
+  /**
+   * AEAD configured while creating an instance of {@link HpkeSuite}
+   *
+   * @return aead
+   */
+  AEAD getAead() {
+    return mAead;
+  }
+
+  enum KEM {
+    DHKEM_X25519_HKDF_SHA256(
+        /* id= */ 32, /* encLength= */ 32, /* pkLength= */ 32, /* skLength= */ 32);
+
+    private final int id;
+    private final int encLength;
+    private final int pkLength;
+    private final int skLength;
+
+    KEM(int id, int encLength, int pkLength, int skLength) {
+      this.id = id;
+      this.encLength = encLength;
+      this.pkLength = pkLength;
+      this.skLength = skLength;
+    }
+
+    /**
+     * KEM id in its decimal representation
+     *
+     * @return kem id
+     * @see <a
+     *         href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-encapsulation-mechanism">KEM
+     *         ids</a>
+     */
+    int getId() {
+      return id;
+    }
+
+    /**
+     * The length in bytes of an encapsulated key produced by this KEM.
+     *
+     * @return encapsulated key size in bytes
+     */
+    int getEncLength() {
+      return encLength;
+    }
+
+    /**
+     * The length in bytes of an encoded public key for this KEM.
+     *
+     * @return public key size in bytes
+     */
+    int getPkLength() {
+      return pkLength;
+    }
+
+    /**
+     * The length in bytes of an encoded private key for this KEM.
+     *
+     * @return private key size in bytes
+     */
+    int getSkLength() {
+      return skLength;
+    }
+
+    /**
+     * Validates the enc size in bytes matches the {@link KEM} spec.
+     *
+     * @param enc encapsulated key produced by the kem
+     * @see <a
+     *         href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-encapsulation-mechanism">expected
+     *         enc size</a>
+     */
+    void validateEnc(byte[] enc) {
+      Preconditions.checkNotNull(enc, "enc");
+      final int expectedLength = this.getEncLength();
+      if (enc.length != expectedLength) {
+        throw new IllegalArgumentException(
+            "Expected enc length of " + expectedLength + ", but was " + enc.length);
+      }
+    }
+
+    /**
+     * Validates the public key size in bytes to match the {@link KEM} public key spec.
+     *
+     * @param publicKey alias pk
+     * @return key in its raw format
+     * @see <a
+     *         href="https://www.rfc-editor.org/rfc/rfc9180.html#name-algorithm-identifiers">expected
+     *         pk size</a>
+     */
+    byte[] validateAndGetPublicKey(PublicKey publicKey) {
```

Comment:
Ditto that this isn't all the validation steps.

Reply:
Same as above. I have updated the name to perform validations only on the length, and updated the name to reflect the raw public key will be extracted from the public key object.